test_scripts/RC/OnRCStatus/commonOnRCStatus.lua

```diff
@@ -0,0 +1,250 @@
+  "SEAT"
+}
+
+function commonOnRCStatus.getRCAppConfig()
+  local struct = commonRC.getRCAppConfig()
+  -- struct.moduleType = { "RADIO", "CLIMATE", "AUDIO", "LIGHT", "HMI_SETTINGS", "SEAT" }
+  struct.moduleType = { "RADIO", "CLIMATE" }
+  return struct
+end
+
+function commonOnRCStatus.getMobileSession(pAppId)
+  return commonRC.getMobileSession(test, pAppId)
+end
+
+function commonOnRCStatus.preconditions()
+  commonRC.preconditions()
+end
+
+function commonOnRCStatus.start()
+  commonRC.start(nil, test)
+end
+
+local function PTUfunc(tbl)
+  local appId = config.application1.registerAppInterfaceParams.appID
+  tbl.policy_table.app_policies[appId] = commonOnRCStatus.getRCAppConfig()
+end
+
+function commonOnRCStatus.RegisterRCapplication(pModuleStatus, ptu_update_func, pAppId)
+  if not pAppId then pAppId = 1 end
+  if not pModuleStatus then pModuleStatus = { freeModules = commonOnRCStatus.modules, allocatedModules = { } } end
+  if not ptu_update_func then
+    ptu_update_func = PTUfunc
+  end
+  commonRC.rai_ptu_n(pAppId, ptu_update_func, test)
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  mobSession:ExpectNotification("OnRCStatus",pModuleStatus)
+  pModuleStatus.appID = commonRC.getHMIAppId()
+  EXPECT_HMINOTIFICATION("RC.OnRCStatus", pModuleStatus )
+end
+
+function commonOnRCStatus.rai_ptu_n(ptu_update_func, pAppId)
+  if not pAppId then pAppId = 1 end
+  if not ptu_update_func then
+    ptu_update_func = PTUfunc
+  end
+  commonRC.rai_ptu_n(pAppId, ptu_update_func, test)
+end
+
+function commonOnRCStatus.rai_n_rc_app(id)
+  commonRC.rai_n(id, test)
+end
+
+function commonOnRCStatus.SubscribeToModule(pModule, pAppId)
+  if not pAppId then pAppId = 1 end
+  commonRC.subscribeToModule(pModule, pAppId, test)
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  mobSession:ExpectNotification("OnRCStatus")
+  :Times(0)
+  EXPECT_HMINOTIFICATION("RC.OnRCStatus")
+  :Times(0)
+end
+
+function commonOnRCStatus.SubscribeToModuleWithoutNot(pModule, pAppId)
+  if not pAppId then pAppId = 1 end
+  commonRC.subscribeToModule(pModule, pAppId, test)
+end
+
+function commonOnRCStatus.unsubscribeToModule(pModule, pModuleStatus, pAppId)
+  if not pAppId then pAppId = 1 end
+  commonRC.unSubscribeToModule(pModule, pAppId, test)
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  mobSession:ExpectNotification("OnRCStatus",pModuleStatus)
+  pModuleStatus.appID = commonRC.getHMIAppId()
+  EXPECT_HMINOTIFICATION("RC.OnRCStatus", pModuleStatus)
+end
+
+function commonOnRCStatus.ActivateApp(pAppId)
+  commonRC.activate_app(pAppId, test)
+end
+
+function commonOnRCStatus.getHMIAppId(pAppId)
+  local appId  = commonRC.getHMIAppId(pAppId)
+  return appId
+end
+
+function commonOnRCStatus.getSettableModuleControlData(pModuleType)
+  local struct = {}
+  if "CLIMATE" == pModuleType or
+    "RADIO" == pModuleType then
+    struct = commonRC.getSettableModuleControlData(pModuleType)
+  elseif pModuleType == "SEAT" then
```

Comment:
@GetmanetsIrina Please remove the logic related to other proposals.
In case they are merged later scripts will be failed.

Reply:
@dboltovskyi , Updated, cases with not implemented modules are not executed for now. Logic is not removed because new modules also must be covered. After implementation common script will be updated according to implemented modules.

Reply:
@GetmanetsIrina Think it's better to use only existing modules for now. Since new modules are not part of this proposal it's better remove the code for them in order to simplify review process.
Later the code can be added again using git history.